front/app/components/admin/GraphCards/ActiveUsersCard/useActiveUsers/index.ts

```diff
@@ -28,58 +24,45 @@ export default function useActiveUsers({
   resolution,
 }: QueryParameters) {
   const { formatMessage } = useIntl();
-
-  const [timeSeries, setTimeSeries] = useState<TimeSeries | NilOrError>();
-  const [stats, setStats] = useState<Stats | NilOrError>();
-  const [xlsxData, setXlsxData] = useState<XlsxData | NilOrError>();
-
-  const [currentResolution, setCurrentResolution] =
-    useState<IResolution>(resolution);
-
-  useEffect(() => {
-    const observable = analyticsStream<Response>(
-      query({
-        projectId,
-        startAtMoment,
-        endAtMoment,
-        resolution,
-      })
-    ).observable;
-
-    const subscription = observable.subscribe(
-      (response: Response | NilOrError) => {
-        if (isNilOrError(response)) {
-          setStats(response);
-          setTimeSeries(response);
-          setXlsxData(response);
-
-          return;
-        }
-
-        const translations = getTranslations(formatMessage);
-
-        setCurrentResolution(resolution);
-
-        const stats = parseStats(response.data);
-        setStats(stats);
-
-        const timeSeries = parseTimeSeries(
-          response.data[0],
-          startAtMoment,
-          endAtMoment,
-          resolution
-        );
-
-        setTimeSeries(timeSeries);
-
-        setXlsxData(
-          parseExcelData(stats, timeSeries, resolution, translations)
-        );
-      }
-    );
-
-    return () => subscription.unsubscribe();
-  }, [projectId, startAtMoment, endAtMoment, resolution, formatMessage]);
+  const [currentResolution, setCurrentResolution] = useState(resolution);
+
+  const { data: analytics } = useAnalytics<Response>(
+    query({
+      projectId,
+      startAtMoment,
+      endAtMoment,
+      resolution,
+    }),
+    () => setCurrentResolution(resolution)
+  );
+
+  const stats = analytics ? parseStats(analytics.data.attributes) : null;
+
+  const timeSeries = useMemo(
+    () =>
+      analytics?.data
+        ? parseTimeSeries(
+            analytics.data.attributes[0],
+            startAtMoment,
+            endAtMoment,
+            currentResolution
+          )
+        : null,
+    [analytics?.data, startAtMoment, endAtMoment, currentResolution]
+  );
+
+  const xlsxData = useMemo(
+    () =>
+      analytics?.data && stats
+        ? parseExcelData(
+            stats,
+            timeSeries,
+            resolution,
```

Comment:
Not blocking but putting `currentResolution` in here as well would be the cherry on the pie